src/blocks/recommendations/class-recommendations-block.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Parse.ly Recommendations Block class file.
+ *
+ * @package Parsely
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely;
+
+/**
+ * Parse.ly Recommendations Block for the WordPress Block Editor.
+ *
+ * @since 3.2.0
+ */
+class Recommendations_Block {
+	const MINIMUM_WORDPRESS_VERSION = '5.6';
+
+	/**
+	 * Determine whether the block and its assets should be registered.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	public function run(): void {
+		global $wp_version;
+
+		if ( ! apply_filters( 'wp_parsely_recommendations_block_enabled', false ) ) {
+			// This block is behind a "feature flag" and it's not enabled. Bail.
+			return;
+		}
+
+		if ( ! isset( $wp_version ) || version_compare( $wp_version, self::MINIMUM_WORDPRESS_VERSION ) < 0 ) {
+			// WordPress is not recent enough to run this block.
+			return;
+		}
+
+		// Temporary workaround - remove block when in FSE due to issues.
+		global $pagenow;
+		if ( 'site-editor.php' === $pagenow ) {
+			add_action(
+				'admin_init',
+				function() {
+					unregister_block_type( 'wp-parsely/recommendations' );
+				}
+			);
+		}
+
+		self::register_block_and_assets();
+	}
+
+	/**
+	 * Registers all block assets so that they can be enqueued through the
+	 * WordPress Block Editor in the corresponding context.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	public static function register_block_and_assets(): void {
+		$plugin_url = plugin_dir_url( PARSELY_FILE );
+
+		$editor_asset_file = require plugin_dir_path( PARSELY_FILE ) . 'build/recommendations-edit.asset.php';
+		wp_register_script(
+			'wp-parsely-recommendations-block-editor',
+			$plugin_url . 'build/recommendations-edit.js',
```

Comment:
What do you think about renaming the file from `recommendations-edit.js` to `recommendations-editor.js`. I think it better represents the intent of the file.

Reply:
For Blocks, creating an `edit.js` and `save.js` file (if needed - we don't have a save file in our case) is a convention, and it's also the way files are generated by the [create-block](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-create-block/) package.

I don't have a strong opinion on renaming the rebuilt file to recommendations-editor, although recommendations-edit is more representative of where the built file comes from. I don't think it matters that much because it's a built file anyway.

Let me know if you want me to go ahead with this.

Reply:
If it's the convention, then let's leave it as-is.